src/internal/rates/protocols/lido/lido.ts

```diff
@@ -0,0 +1,74 @@
+import { Provider } from '@ethersproject/abstract-provider';
+import { Signer } from '@ethersproject/abstract-signer';
+import { BigNumber, Contract } from 'ethers';
+import { SECONDS_PER_YEAR } from '../../../../constants/index.js';
+import { unwrap } from '../../../../helpers/result.js';
+import { fixed } from '../../helpers/index.js';
+import { BASIS_POINTS, LIDO_ABI, LIDO_ORACLE_ABI, WSTETH_ABI } from './constants.js';
+import { LidoContract, LidoOracleContract, wstETHContract } from './types.js';
+
+/**
+ * Lido's exchangeRate implementation.
+ *
+ * @remarks
+ * https://docs.lido.fi/contracts/wsteth#stethpertoken
+ *
+ * A Lido market's `cTokenAddr` is the wstETH address.
+ *
+ * @internal
+ */
+export async function exchangeRateLido (a: string, s: Provider | Signer): Promise<string | undefined> {
+
+    const contract = new Contract(a, WSTETH_ABI, s) as wstETHContract;
+
+    return (await contract.stEthPerToken()).toString();
+}
+
+/**
+ * Lido's interestRate implementation.
+ *
+ * @remarks
+ * https://docs.lido.fi/contracts/lido-oracle#add-calculation-of-staker-rewards-apr
+ *
+ * A Lido market's `cTokenAddr` is the wstETH address.
+ *
+ * @internal
+ */
+export async function interestRateLido (a: string, s: Provider | Signer): Promise<string | undefined> {
+
+    const wstETH = new Contract(a, WSTETH_ABI, s) as wstETHContract;
+
+    // get the lido address from the wstETH contract
+    const lidoAddress = await wstETH.stETH();
+
+    const lido = new Contract(lidoAddress, LIDO_ABI, s) as LidoContract;
+
+    // get the fee and the oracle address from the lido contract
+    const [fee, oracleAddress] = await Promise.all([
+        lido.getFee(),
+        lido.getOracle(),
```

Comment:
Nice this was my first question, good that we can grab the oracle address from the stETH contract